### PR TITLE
ZKP section

### PR DESCRIPTION
**Pull request by brentzundel**
Addresses issue #24 
Signed-off-by: Brent <brent.zundel@gmail.com>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1521,6 +1479,55 @@ <h3>Hashlinks</h3>
...
     </section>
 
+    <section>
+      <h2>Zero-Knowledge Proofs</h2>
+      <em>This section is non-normative</em>
+        <p>
+The Verifiable Credentials Data Model is designed to be compatible with a
+variety of existing and emerging digital proof formats. Each
+proof format has benefits and drawbacks. Many proof formats cannot reveal
+selected attribute values in a verifiable credential; they can only reveal all
```
Comment (TallTed) [inline index.html]: `selected attribute values in a`
-> `only selected attribute values from a`

(add `only`; change `in` to `from`)

File: index.html [hunk 2]
```diff
@@ -1521,6 +1479,55 @@ <h3>Hashlinks</h3>
...
+(or none).
+        </p>
+        <p>
+Zero-Knowledge Proofs (ZKPs) are a proof format that enables
+privacy-preserving and data-minimization features in
```
Comment (TallTed) [inline index.html]: `privacy-preserving and data-minimization`
-> `privacy-preservation and data-minimization`
or
-> `privacy-preserving and data-minimizing`

Whether `ation` or `ing` is chosen here, text throughout should also be changed to match.

File: index.html [hunk 3]
```diff
@@ -1521,6 +1479,55 @@ <h3>Hashlinks</h3>
...
+        </p>
+        <p>
+Zero-knowledge predicate proofs are a type of proof associated with
+an attribute. Predicate proofs include comparisons, such as
+"greater than", "less than", "not equal", "range",
+"set membership", and "set non-membership".
+Predicate proofs can be constructed without requiring the issuer
+to create special fields, such as
+"age-under-18", "age-over-21", or "age-over-60", in the verifiable credential
+at the time the credential was issued.
```
Comment (TallTed) [inline index.html]: `was` 
-> `is`

(or rephrase the earlier clause, changing `without requiring the issuer to create` to `without requiring the issuer to have created`)

File: index.html [hunk 4]
```diff
@@ -1521,6 +1479,55 @@ <h3>Hashlinks</h3>
...
+the verifiable credential can cryptographically prove to the verifier that
+the holder's age is greater (or less) than a specified number, without
+revealing the holder's "birth-date".
+        </p>
+        <p>
+Verifiable credentials based on zero-knowledge proof mechanisms are
+also quantum-resistant after presentation.
+        </p>
+        <p>
+A drawback of zero-knowledge proof mechanisms is that they are
```
Comment (TallTed) [inline index.html]: `A drawback of zero-knowledge proof mechanisms is that they are`
-> `Drawbacks of zero-knowledge proof mechanisms include that they are`

File: index.html [hunk 5]
```diff
@@ -1521,6 +1479,55 @@ <h3>Hashlinks</h3>
...
+more complex and larger than some older signature schemes.
```
Comment (TallTed) [inline index.html]: `more complex and larger`
reads better as `larger and more complex`

File: index.html (hunk 2 above)
Comment (brentzundel) [reply index.html]: Would this work?
```
enables
privacy-preserving data-minimization features . . .
```

Comment (TallTed) [reply index.html]: That would work for this sentence. The rest of the doc should still be checked for similar disagreements.

File: index.html (hunk 5 above)
Comment (brentzundel) [reply index.html]: re-worded in [5ff637c](https://github.com/w3c/vc-imp-guide/pull/32/commits/5ff637c784e5bec18463221499a55af27f8746e1)

File: index.html (hunk 4 above)
Comment (brentzundel) [reply index.html]: fixed in [5ff637c](https://github.com/w3c/vc-imp-guide/pull/32/commits/5ff637c784e5bec18463221499a55af27f8746e1)

File: index.html (hunk 3 above)
Comment (brentzundel) [reply index.html]: fixed in [5ff637c](https://github.com/w3c/vc-imp-guide/pull/32/commits/5ff637c784e5bec18463221499a55af27f8746e1)

File: index.html (hunk 2 above)
Comment (brentzundel) [reply index.html]: fixed in [5ff637c](https://github.com/w3c/vc-imp-guide/pull/32/commits/5ff637c784e5bec18463221499a55af27f8746e1)

File: index.html (hunk 1 above)
Comment (brentzundel) [reply index.html]: fixed in [5ff637c](https://github.com/w3c/vc-imp-guide/pull/32/commits/5ff637c784e5bec18463221499a55af27f8746e1)

**Review by ken-ebert — APPROVED** [review]

**Review by TallTed — APPROVED** [review]

**Pull request merged**